tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/BasicAuthHandler.java

```diff
@@ -20,33 +20,39 @@
  * =====
  */
 
-import org.apache.http.HttpResponse;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpUriRequest;
-import org.apache.http.client.methods.RequestBuilder;
-import org.apache.http.entity.ContentType;
-import org.apache.http.entity.StringEntity;
-import org.apache.http.impl.client.HttpClientBuilder;
-
 import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
 import java.util.HashMap;
 import java.util.Map;
 
 public class BasicAuthHandler {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
-    public static String auth(String baseUrl, TaskParams.BasicAuth auth) throws IOException {
+    private BasicAuthHandler() { }
+
+    public static String auth(HttpClient.Builder httpBuilder, String baseUrl, TaskParams.BasicAuth auth) throws IOException {
         Map<String, Object> body = new HashMap<>();
```

Comment:
`var` here? Should var usage be at least method-wide or, better yet, for the entire file (unless it requires casting like in the `lastError` case below)?